main/src/library/Array.flix

```diff
@@ -105,7 +105,8 @@ namespace Array {
     ///
     /// Equivalent to the expression `a[b..e]`.
     ///
-    pub def slice(b: Int32, e: Int32, a: Array[a, r]): Array[a, r] \ { Read(r), Write(r) } = a[b..e]
+    pub def slice(_: Region[r1], b: Int32, e: Int32, a: Array[a, r]): Array[a, r1] \ { Read(r), Write(r1) } =
+        unsafe_cast a[b..e] as Array[a, r1] \ { Read(r), Write(r1) }
```

Comment:
This will be refactored, but in a separate PR.

Reply:
Thanks - am I reverting the code?

Reply:
No its fine for now :)

Reply:
Oh yeah - it's already been merged.